src/reactionsystem.jl

```diff
@@ -0,0 +1,165 @@
+using Catalyst, ModelingToolkit, Symbolics
+
+# module SBML
+# struct Model end
+# struct Reaction end
+# struct Compartment end
+# struct Species end
+# end
+
+""" ReactionSystem constructor """
+function ModelingToolkit.ReactionSystem(model::Model; kwargs...)  # Todo: requires unique parameters (i.e. SBML must have been imported with localParameter promotion in libSBML)
+    model = make_extensive(model)
+    # model = expand_reversible(model)
+    rxs = mtk_reactions(model)
+    t = Catalyst.DEFAULT_IV
+    species = [create_var(k) for k in keys(model.species)]
+    params = vcat([create_param(k) for k in keys(model.parameters)], [create_param(k) for k in keys(model.compartments)])
+    ReactionSystem(rxs,t,species,params; kwargs...)
+end
+
+""" ReactionSystem constructor """
+function ModelingToolkit.ReactionSystem(sbmlfile::String; kwargs...)
```

Comment:
I'm not sure we will want to steal the String dispatch, since there are other formats that create a ReactionSystem (BioNetGen IIRC). 

I think just having `ModelingToolkit.ReactionSystem(model::Model; kwargs...)` might be good enough.
It's two function calls, but saves us from committing type piracy down the line.